File: solutions_1483485_0/java/rogerfgm/GA.java
import java.io.*;
import java.util.*;




public class GA {

	String s  = null;
	String[] sp = null;
	
	
	
	public void run() throws Exception{
		
		BufferedReader br = new BufferedReader(new FileReader("A-small-attempt1.in"));
		
		BufferedWriter bw = new BufferedWriter(new FileWriter("OUTPUT.txt"));
		s = br.readLine();
		int T = Integer.parseInt(s);
		
		String[] as = {"ejp mysljylc kd kxveddknmc re jsicpdrysi", "rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd", "de kr kd eoya kw aej tysr re ujdr lkgc jv"};
		String[] bs = {"our language is impossible to understand", "there are twenty six factorial possibilities", "so it is okay if you want to just give up"};
		
		Map<String, String> map = new HashMap<String, String>();
		
		for(int i = 0; i < as.length; i++){
			String a = as[i];
			String b = bs[i];
			for(int j = 0; j < a.length(); j++){
				String sa = a.substring(j, j+1);
				String sb = b.substring(j, j+1);
				map.put(sa, sb);
			}
		}
		
		map.put("q", "z");
		map.put("y", "a");
		map.put("e", "o");
		
		for(char c = 'a'; c <= 'z'; c++){
			String sa = Character.toString(c);
			if(!map.containsKey(sa)){
				Collection<String> ks = map.values();
				for(char cs = 'a'; cs <= 'z'; cs++){
					String sb =  Character.toString(cs);
					if(!ks.contains(sb)){
						map.put(sa, sb);
						break;
					}
				}
			}
		}
		
		
		
		int t = 1;
		while(t <= T){
			s = br.readLine();
			StringBuilder ans = new StringBuilder();
			
			for(int i = 0; i < s.length(); i++){
				String sa = s.substring(i, i+1);
				String sb = map.get(sa);
				ans.append(sb);
			}
			
			bw.write("Case #" + t + ": " + ans + "\n");
			
			t++;
		}
		
		bw.close();
	}
	

	
	/**
	 * @param args
	 */
	public static void main(String[] args) throws Exception {
		// TODO Auto-generated method stub
		GA b = new GA();
		b.run();

		
	}
	
	

}

File: solutions_1483485_0/C++/tuananh93/ga.cpp
#include <iostream>
#include <cstdio>
#include <cstdlib>
#include <algorithm>
#include <cmath>
#include <cstring>
#include <string>
#include <map>
#include <set>
#include <stack>
#include <list>
#include <vector>
#include <queue>

using namespace std;

#define PI acos(-1)
#define MP make_pair
#define PB push_back
#define VI vector <int>
#define PII pair <int, int>
#define LL long long
#define SET(v,i) memset(v, i, sizeof(v))
#define FOR(i,a,b) for (int i = (a); i <= (b); i++)
#define FORD(i,a,b) for (int i = (a); i >= (b); i--)
#define FORN(i,a,b) for (int i = (a); i < (b); i++)
#define DOWN(i,a,b) for (int i = (a); i > (b); i--)
#define FIT(it,v) for (typeof(v.begin()) it = v.begin(); it != v.end(); it++)
#define FITD(it,v) for (typeof(v.rbegin()) it = v.rbegin(); it != v.rend(); it++)
#define FREOPEN freopen("A-small-attempt2.in", "r", stdin); freopen("a.out", "w", stdout)

char s[110];
map <char, char> f;

string a[6] = { "ejp mysljylc kd kxveddknmc re jsicpdrysi",
				"our language is impossible to understand",
				"rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd",
				"there are twenty six factorial possibilities",
				"de kr kd eoya kw aej tysr re ujdr lkgc jv",
				"so it is okay if you want to just give up" };
				
int main() {
	int t;
	FREOPEN;
	f['a'] = 'y';
	f['z'] = 'q';
	f['o'] = 'e';
	f[' '] = ' ';
	
	FOR (i, 0, 5)
	if ((i % 2) == 0) {
		int j = 0;
		while (a[i][j]) {
			f[a[i][j]] = a[i + 1][j];
			j++;
		}
	}
	for (char c = 'a'; c <= 'z'; c++) {
		int ok = 1;
		for (char x = 'a' ; x <= 'z'; x++)
			if (f[x] == c) ok = 0;
		if (ok == 1) f['q'] = c;
	}
	
	cin >> t; gets(s);
	
	FOR (test, 1, t) {
		gets(s);
		int i = 0;
		//printf("%s\n", s);
		while (s[i]) {
			s[i] = f[s[i]];
			//cerr << s[i] << endl;
			i++;
		}
		printf("Case #%d: %s\n", test, s);		
	}
	return 0;
}
